### DISCUSSION TITLE
gptel and LMStudio

### DISCUSSION THREAD

**Discussion by cyurek** [karthink/gptel#392] [Q&A]
Is gptel compatible with LMStudio and if it is what would be the elisp code to set it up?

**Comment by karthink** [comment]
Not out of the box.  If LMStudio offers an http API I can add support.

**Reply by cyurek** [reply]
You can use LLMs you load within LM Studio via an API server running on localhost. Requests and responses follow OpenAI's API format. Point any code that currently uses OpenAI to localhost:PORT to use a local LLM instead.

**Reply by karthink** [reply]
Then you should be able to use gptel with it, see uses of `gptel-make-openai` in the README for examples.

**Reply by hierophantos** [reply]
> Then you should be able to use gptel with it, see uses of `gptel-make-openai` in the README for examples.

Currently this is found [here](https://github.com/karthink/gptel?tab=readme-ov-file#privategpt) in the README under the function `gptel-make-privategpt`. Works great! 🙌
